socorro/external/es/crashstorage.py

```diff
@@ -211,6 +211,77 @@ def save_raw_and_processed(self, raw_crash, dumps, processed_crash,
 
 
 #==============================================================================
+class ESCrashStorageRestrictedDump(ESCrashStorage):
+    """This class stores redacted crash reports into Elasticsearch, but instead
+    of removing the entire `json_dump`, it keeps only a subset of its keys.
+    """
+    required_config = Namespace()
+    required_config.add_option(
+        name="json_dump_keys",
+        doc="keys of the json_dump field to keep in the processed crash",
+        default=[
+            "largest_free_vm_block",
+            "tiny_block_size",
+            "write_combine_size",
+        ],
+        from_string_converter=list_converter,
+    )
+
+    required_config.namespace('es_redactor')
+    required_config.es_redactor.add_option(
+        name="redactor_class",
+        doc="the name of the class that implements a 'redact' method",
+        default='socorro.external.crashstorage_base.Redactor',
+        from_string_converter=class_converter,
+    )
+    required_config.es_redactor.forbidden_keys = change_default(
+        Redactor,
+        "forbidden_keys",
+        "upload_file_minidump_flash1.json_dump, "
+        "upload_file_minidump_flash2.json_dump, "
+        "upload_file_minidump_browser.json_dump"
+    )
```

Comment:
There's something "unnatural" about this. "Redacting" and "forbidden keys" is something that fits with the definition of the word "redact" ("censor or obscure (part of a text) for legal or security purposes.")

It just feels like we're mixing things a bit too wildly. I thought we used this redacting business for the sake of security. Not for the sake of dieting. Is that a concern you share?